contributing.md

```diff
@@ -0,0 +1,163 @@
+
+Or, equivalently:
+
+    $ git clone https://github.com/beetbox/beets.git
+    $ cd beets
+    $ pip install -e .
+
+If you already have a released version of beets installed, you may need to
+remove it first by typing `pip uninstall beets`. The pip command above will put
+the beets source in a `src/beets` directory and install the `beet` CLI script to
+a standard location on your system. You may want to use the `--src` option to specify
+the parent directory where the source will be checked out and the `--user` option
+such that the package will be installed to your home directory (compare with the output of
+`pip install --help`). 
+
+Alternatively, you can get the source via 
+[git at GitHub](http://github.com/beetbox/beets). The pip method above installs
+dependencies automatically, but if you do it manually, you can use pip to
+install these Python modules:  
+* `mutagen`
+* `munkres`
+* `unidecode`
+* `pyYAML`
+* `musicbrainzngs`
+* `confuse`
+* `mediafile`
+* `jellyfish`
+
+Please be aware that this list might be out of date, but [setup.py](https://github.com/beetbox/beets/blob/60bba370c0ecc58b4057d78867273e5936d22dee/setup.py#L89-L110) contains a list that is always up to date. With all the required modules in place, you can just type `./beet` in the source directory to run beets from there.
+
+
+### Code contribution ideas
+* We maintain a set of [issues marked as "bite-sized"](https://github.com/beetbox/beets/labels/bitesize). These are issues that would serve as a good introduction to the codebase. Claim one and start exploring!
+* Our [test coverage](https://codecov.io/github/beetbox/beets) is somewhat low. You can help out by finding low-coverage modules on Coveralls and adding new tests.
+* There are several ways to improve the tests in general (see [[Testing]]) and some places to think about performance optimization (see [[Optimization]]).
+* Not all of our code is up to our coding conventions. In particular, the [API documentation][apidocs] are currently quite sparse. You can help by adding to the docstrings in the code and to the documentation pages themselves.
+
+## Your first contribution
+If this is your first time contributing to an open source project, welcome! If you are confused at all about how to contribute or what to contribute, take a look at [this great tutorial](http://makeapullrequest.com/), or stop by our [gitter chat room](https://gitter.im/beetbox/beets#) if you have any questions. 
+
+We maintain a list of issues we feel are perfect for first-timers tagged "[good first issue](https://github.com/beetbox/beets/labels/good%20first%20issue)". These issues are perfect for anyone new to open source. In fact, we encourage only users who are new to open source to tackle these. Ideally, the goal for these issues is to get users comfortable with contributing to an open source project for the first time. Please, do not hesitate to ask any questions.
```

Comment:
So I mentioned `first-timers-only` on discourse, but I believe this is a good enough equivalent. Basically, imo, the philosophy should be issues where the only real thinking should be how to create a fork and pull request. Ideally, any issue issue with this tag would be restricted (or heavily encouraged) to only be tackled by users who are first-timers to open source projects. Additionally, any issues with this tag would outline in detail what steps the user should take to correct the issue to the extent of what files or lines to change.

Reply:
Cool, that sounds good. I have no strong feelings about the name, certainly, but I like this idea of scoping these entry-level issues so they're appropriate for people who want to learn how to make a PR in the first place, for example. Maybe we want a separate tag for smallish issues that are more appropriate for experienced hackers, but maybe that's not really a significant category.

Reply:
That's actually what I think of when I think of `byte-sized`. Not necessarily any specific difficulty, but small in size. I think the more labels you create, the harder it can be to keep track of them while remaining consistent in using them, and more time gets spent on considering the difficulty and size of issues. Whether or not that's worthwhile, I'll leave up to you.